src/Present.hs

```diff
@@ -132,6 +133,12 @@ normalizeType = go
             TH.StarT -> fail "Star (*) is not supported."
             TH.ConstraintT -> fail "Constraints are not supported."
             TH.LitT _ -> fail "Type-level literals are not supported."
+#if MIN_VERSION_template_haskell(2,11,0)
+            TH.InfixT{} -> fail "Infix type constructors are not supported."
+            TH.UInfixT{} -> fail "Unresolved infix type constructors are not supported."
+            TH.ParensT _ -> fail "Parenthesized types are not supported."
+            TH.WildCardT -> fail "Wildcard types are not supported."
```

Comment:
> They're left's:

I'm not convinced! ;)

```
λ> fail "Bla" :: Either String Int
*** Exception: Bla
```

Also, see http://hackage.haskell.org/package/base-4.9.0.0/docs/Data-Either.html#v:fail